buffer/src/main/java/io/netty/buffer/Unpooled.java

```diff
@@ -262,38 +262,86 @@ public static ByteBuf wrappedBuffer(ByteBuffer... buffers) {
         return wrappedBuffer(buffers.length, buffers);
     }
 
+    // used to avoid temporary array allocation/copy when wrapping byte[][] and ByteBuffer[]
+    abstract static class WrappingIterable<T> implements Iterable<ByteBuf> {
+        private final T[] array;
+        private final int start;
+
+        protected WrappingIterable(T[] array) {
+            this.array = array;
+
+            for (int i = 0;; i++) {
+                if (i >= array.length || array[i] == null) {
+                    start = array.length;
+                    return;
+                }
+                if (!isEmpty(array[i])) {
+                    start = i;
+                    return;
+                }
+            }
+        }
+
+        int maxSize() {
+            return array.length - start;
+        }
+
+        @Override
+        public Iterator<ByteBuf> iterator() {
+            return new Iterator<ByteBuf>() {
+                int i = start;
+
+                @Override
+                public boolean hasNext() {
+                    for (; i < array.length && array[i] != null; i++) {
+                        if (!isEmpty(array[i])) {
+                            return true;
+                        }
+                    }
+                    return false;
+                }
+
+                @Override
+                public ByteBuf next() {
+                    if (hasNext()) {
+                        return wrap(array[i++]);
+                    }
```

Comment:
You could make this a bit more efficient by merge some code of `hasNext` in here you will need to do two array access here.

Reply:
@njhill ...

Reply:
@normanmaurer i think this may be a github quirk - it didn't update the PR front page with the change, but I did make it (actually refactored the iterator a bit more to address this) - check the changes tab/commits